src/bindgen/ir/ty.rs

```diff
@@ -17,6 +17,55 @@ use crate::bindgen::monomorph::Monomorphs;
 use crate::bindgen::utilities::IterHelpers;
 use crate::bindgen::writer::{Source, SourceWriter};
 
+// Struct that serves as key for resolving associated types
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct AssocTypeId{
+    pub ty: Type,
+    pub trait_: GenericPath,
+    pub ident: String,
+}
+
+impl AssocTypeId{
+    
+    pub fn new(type_path: &syn::TypePath) -> Result<Self, String>{
+        
+        let self_type = if let Some(qself) = &type_path.qself{
+            qself.ty.as_ref()
```

Comment:
```suggestion
            qself.ty
```

this should be sufficient

Reply:
This causes the compiler to complain that we cannot move out of a shared reference since the expression to destructure is a reference. I managed to achieve the same effect with "&qself.ty", would that be better?

Reply:
yes, that's better